File: tutorial/Event.py
from AnalysisTopGNN.Templates import EventTemplate
from Particles import Jet, Top

class Event(EventTemplate):

    def __init__(self):
        EventTemplate.__init__(self)
        
        # ===== Event Variable Declaration ===== # 
        self.weight = "weight_mc"
        self.mu = "mu"
        self.met = "met_met"
        self.phi = "met_phi"

        self.Trees = ["nominal"]

        self.Objects = {
                "SomeJets" : Jet(), 
                "Tops" : Top()
                }
        # ===== End of declaration ===== #
        self.DefineObjects()
        self._CommitHash = "<Some Hash For Bookkeeping>"
    
    def CompileEvent(self): 
        # Create njets variable for the event 
        self.nJets = len(self.SomeJets) 
        
        # Do some particle truth matching 
        for i in self.SomeJets:
            index = self.SomeJets[i].MatchedTops
            if index[0] == -1:
                continue 
            for t in index:
                self.SomeJets[i].Parent.append(self.Tops[t])

        # (Optional) Convert the dictionary to particle list.
        self.SomeJets = self.DictToList(self.SomeJets)
        self.Tops = self.DictToList(self.Tops)

File: analysis-rebuild/src/analysisg/events/bsm_4tops/event.cxx
#include "event.h"

bsm_4tops::bsm_4tops(){
    this -> name = "bsm_4tops"; 
    this -> add_leaf("weight", "weight_mc"); 
    this -> add_leaf("event_number", "eventNumber"); 
    this -> add_leaf("mu", "mu"); 
    this -> add_leaf("met", "met_met"); 
    this -> add_leaf("phi", "met_phi"); 
    this -> trees = {"nominal"}; 

    this -> register_particle(&this -> Tops);
    this -> register_particle(&this -> Children); 
    this -> register_particle(&this -> TruthJets);
    this -> register_particle(&this -> Jets); 
    this -> register_particle(&this -> Partons); 
}

bsm_4tops::~bsm_4tops(){}

event_template* bsm_4tops::clone(){return (event_template*)new bsm_4tops();}

void bsm_4tops::build(element_t* el){
    el -> get("event_number", &this -> event_number); 

}

